test/config.js

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+import os from 'os';
+import test from 'ava';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+const homedirStub = sinon.stub(os, 'homedir');
+const fixtureBasePath = path.resolve('test', 'fixtures', 'config');
+
+test('should return config from `.np-config.json` in home-directory when global binary used', async t => {
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir1'));
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': true,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: false});
+});
+
+test('should return config from `.np-config.js` in home-directory when global binary used', async t => {
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir2'));
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': true,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: true, contents: 'dist'});
+});
+
+test('should return config from `package.json` when local binary used', async t => {
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': false,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: true});
```

Comment:
Maybe add the `homedir` stub here as well, for added security?

Reply:
In hindsight, I guess this is pretty much same as the next test.

Reply:
I removed all redundancies in the current implementation.